src/main/java/com/xpanxion/java/springboot/da1/demo/service/student3/GymService3.java

```diff
@@ -45,6 +46,46 @@
 
     }
 
+    public WorkoutTime getWorkoutTime(int memberId, String type){
+        List<WorkoutTime> workoutTimeList = new ArrayList<WorkoutTime>();
+
+        String pattern = "yyyy-MM-dd";
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(pattern);
+
+        var l=  getWorkoutHistory(memberId); // sorted
+
+        Date checkinTime = null;
+        Date checkoutTime;
+        String workoutDate;
+        ParsePosition pp1 = new ParsePosition(0);
+
+        for(WorkoutHistoryPresentation w : l){
+            if (w.getCheckType3().equals(CheckType3.CHECKIN)){
+                checkinTime = w.getTime();
+            }else if (w.getCheckType3().equals(CheckType3.CHECKOUT)){
+                checkoutTime = w.getTime();
+                var deltaMs = checkoutTime.getTime() - checkinTime.getTime();
+                var deltaMinutes = deltaMs / 60000;
```

Comment:
`60000` These are known as "magic numbers".  We typically want to avoid these and define them in a constant at the top of the class... something like `final int MINUTES_SECONDS` or whatever.